### PR TITLE
PMM-12710:  API breaking changes

### PR DESCRIPTION
**Pull request by doracretu3pillar**
Jira ticket: 
https://perconadev.atlassian.net/browse/PMM-12710

### CODE REVIEW

File: pmm-app/src/pmm-qan/panel/components/Details/Details.service.ts [hunk 1]
```diff
@@ -14,7 +14,7 @@ const getExample = async ({
     tables,
   };
 
-  return apiRequestQAN.post<QueryExampleResponse, any>('/ObjectDetails/GetQueryExample', data);
+  return apiRequestQAN.post<QueryExampleResponse, any>('/v1/qan/query:getExample', data);
```
Comment (matejkubinec) [inline pmm-app/src/pmm-qan/panel/components/Details/Details.service.ts:17]: I think the version shouldn't be specified here, but in the `components/helpers/api.ts`. Specifying it here results in:

```
http://localhost/v0/qan/v1/qan/metrics:getFilters
```

File: setup-page/src/App.jsx [hunk 2]
```diff
@@ -16,16 +16,13 @@ export function App() {
     setLoading(true);
 
     try {
-      const response = await fetch('/v1/AWSInstanceCheck', {
-        method: 'POST',
+      const response = await fetch(`/v1/server/AWSInstance?instance_id${instanceId.trim()}`, {
```
Comment (matejkubinec) [inline setup-page/src/App.jsx:19]: Why is the trim needed here?

File: pmm-app/src/pmm-qan/panel/components/Details/Details.service.ts (hunk 1 above)
Comment (YashSartanpara1) [reply pmm-app/src/pmm-qan/panel/components/Details/Details.service.ts]: Updated.

File: pmm-app/src/pmm-qan/panel/components/Details/database-models/mongodb/service.ts [hunk 3]
```diff
@@ -1,7 +1,7 @@
-import { apiRequestManagement } from 'shared/components/helpers/api';
+import { apiRequestActions } from 'shared/components/helpers/api';
 
 export default {
   getTraditionalExplainJSONMongo(body) {
-    return apiRequestManagement.post<any, any>('/Actions/StartMongoDBExplain', body);
+    return apiRequestActions.post<any, any>(':startServiceAction', body);
```
Comment (matejkubinec) [inline pmm-app/src/pmm-qan/panel/components/Details/database-models/mongodb/service.ts:5]: I haven't checked, but the body might need to be adjusted (seeing as the different StartTECHExplain merged into one).

File: pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/service.ts [hunk 4]
```diff
@@ -1,23 +1,7 @@
-import { apiRequestManagement } from 'shared/components/helpers/api';
+import { apiRequestActions } from 'shared/components/helpers/api';
 
 export default {
-  getShowCreateTableMySQL(body) {
```
Comment (matejkubinec) [inline pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/service.ts:4]: One option is to leave the service functions and adjust the payload in the service function bodies to match what the BE expects now.

File: pmm-app/src/shared/core/Settings.service.ts [hunk 5]
```diff
@@ -3,8 +3,8 @@ import { API } from './constants';
 import { Settings, SettingsAPIResponse, SettingsPayload } from './types';
 
 export const SettingsService = {
-  async getSettings(disableNotifications = false): Promise<Settings> {
-    const { settings } = await apiRequest.post(API.SETTINGS, {}, disableNotifications) as SettingsAPIResponse;
+  async getSettings(): Promise<Settings> {
+    const { settings } = await apiRequest.get(API.SETTINGS) as SettingsAPIResponse;
```
Comment (matejkubinec) [inline pmm-app/src/shared/core/Settings.service.ts:7]: Please check the usage of settings as the api response changed (there are no longer enable/disable for each of the setting items only enable)

File: pmm-app/src/pmm-qan/panel/components/Details/database-models/mongodb/service.ts (hunk 3 above)
Comment (YashSartanpara1) [reply pmm-app/src/pmm-qan/panel/components/Details/database-models/mongodb/service.ts]: Updated.

File: pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/service.ts (hunk 4 above)
Comment (YashSartanpara1) [reply pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/service.ts:4]: Updated.

File: pmm-app/src/shared/core/Settings.service.ts (hunk 5 above)
Comment (YashSartanpara1) [reply pmm-app/src/shared/core/Settings.service.ts:7]: Fixed.

File: pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts [hunk 6]
```diff
@@ -80,8 +80,8 @@ export const fetchExplains = async (
       };
 
       const [classicResult, jsonResult] = await Promise.all([
-        mysqlMethods.getExplainTraditional(payload).then(getActionResult),
-        mysqlMethods.getExplainJSON(payload).then(getActionResult),
+        mysqlMethods.getExplainMysql(payload).then(getActionResult),
```
Comment (matejkubinec) [inline pmm-app/src/pmm-qan/panel/components/Details/Explain/Explain.tools.ts:83]: The `MySQL` in the method name is probably redundant as the `mysqlMethods` already provides information that the explain methods are for mysql.

File: pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/mysql.ts [hunk 7]
```diff
@@ -44,11 +44,11 @@ export const mysqlMethods = {
     return result.mysql_show_table_status.action_id;
   },
 
-  getExplainJSON: async ({ example, queryId, placeholders }) => {
+  getExplainJSONMysql: async ({ example, queryId, placeholders }) => {
     try {
       const payload = getExplainPayload(example, queryId, placeholders);
 
-      const result = await MysqlDatabaseService.getTraditionalExplainJSONMysql(payload);
+      const result = await MysqlDatabaseService.getExplainJSONMysql(payload);
```
Comment (matejkubinec) [inline pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/mysql.ts:51]: Same as with the `mysqlmethods`, the mysql in the method name is redundant.

Comment (YashSartanpara1) [reply pmm-app/src/pmm-qan/panel/components/Details/database-models/mysql/mysql.ts]: Updated.

**Review by matejkubinec — APPROVED** [review]

**Pull request merged**